airflow/providers/docker/decorators/docker.py

```diff
@@ -0,0 +1,159 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import base64
+import inspect
+import os
+import pickle
+import shlex
+from tempfile import TemporaryDirectory
+from textwrap import dedent
+from typing import Callable, Dict, Optional, TypeVar
+
+from airflow.decorators.base import DecoratedOperator, task_decorator_factory
+from airflow.providers.docker.operators.docker import DockerOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.utils.python_virtualenv import remove_task_decorator, write_python_script
+
+
+def _generate_decode_command(env_var, file):
+    return shlex.quote(
+        f'python -c "import os; import base64;'
+        f' x = base64.b64decode(os.environ["{env_var}"]);'
+        f' f = open("{file}", "wb"); f.write(x);'
```

Comment:
Theoratically `env_var` and `file` could contain the quote character (`"`), so it’s better to use `!r` instead.